### PR TITLE
[ENH] Scatter Plot: Improve tooltips

### PR DESCRIPTION
**Pull request by jerneju**
##### Description of changes

- Tooltip shows axes variables first and then classes, metas and attributes (in that order).
- Selected variables are no longer shown twice. Number of variables shown is limited.
- The overall look is improved by using proper HTML and some bold face fonts.
- Test for tooltips are added.

##### Includes
- [X] Code changes
- [X] Tests

### CODE REVIEW

**Comment by jerneju** [comment]
@astaric @janezd @kernc Please post some comments.

File: Orange/widgets/visualize/owscatterplotgraph.py [hunk 1]
```diff
@@ -1150,30 +1150,36 @@ def help_event(self, event):
...
-                            len(self.domain.attributes) - 2)
-                if self.domain.class_var:
-                    text += 'Class:\n   {} = {}\n'.format(
-                        self.domain.class_var.name,
-                        self.data[index][self.data.domain.class_var])
+                for v in set([self.shown_x, self.shown_y]):
+                    text += '   {} = {}\n'.format(v, self.data[index][v])
+                if self.tooltip_shows_all:
+                    for columns in (CLASS, META, ATTRIBUTE):
+                        text = show_it(text, columns)
```
Comment (kernc) [inline Orange/widgets/visualize/owscatterplotgraph.py]: Unpack: `*columns`.

File: Orange/widgets/visualize/owscatterplotgraph.py [hunk 2]
```diff
@@ -1150,30 +1150,36 @@ def help_event(self, event):
         if self.scatterplot_item is None:
             return False
 
+        domain = self.data.domain
+        CLASS = ("Class", "Classes", 4, domain.class_vars)
+        META = ("Meta", "Metas", 4, domain.metas)
+        ATTRIBUTE = ("Attribute", "Attributes", 10, domain.attributes)
+
+        def show_it(text, columns):
```
Comment (kernc) [inline Orange/widgets/visualize/owscatterplotgraph.py]: Take `columns` spec unpacked.

File: Orange/widgets/visualize/owscatterplotgraph.py [hunk 3]
```diff
@@ -1150,30 +1150,36 @@ def help_event(self, event):
...
+            n_cols = len(cols)
+            if n_cols:
+                text += "{}:\n".format(columns[len(cols) > 1])
+            text += "".join(
+                '   {} = {}\n'.format(var.name, self.data[index][var]) for var in cols[:columns[2]])
+            if n_cols == columns[2] + 1:
+                var = cols[columns[2]]
+                text += "".join('   {} = {}\n'.format(var.name, self.data[index][var]))
+            elif n_cols > columns[2]:
+                text += "   ... and {} others\n\n".format(n_cols - columns[2])
```
Comment (kernc) [inline Orange/widgets/visualize/owscatterplotgraph.py]: This code is an insult to this craft.

HTH

**Comment by janezd** [comment]
I can't merge this, it's mostly my code. Can someone else take a look, please?

File: Orange/widgets/visualize/owscatterplotgraph.py [hunk 4]
```diff
@@ -1174,36 +1174,46 @@ def help_event(self, event):
...
+                + "<br/>".join(cols)
+
+        def point_data(p):
+            point_data = self.data[p.data()]
+            text = "<br/>".join(
+                format_val(var, point_data, bold=self.tooltip_shows_all)
+                for var in (self.shown_x, self.shown_y))
+            if self.tooltip_shows_all:
+                text += "<br/><br/>" + \
+                        "<br/>".join(show_part(point_data, *columns)
```
Comment (lanzagar) [inline Orange/widgets/visualize/owscatterplotgraph.py]: I suggest joining with "" here and removing one <br/> in the above line. And instead adding the <br/> at the beginning of line 1199.

Currently, if show_part returns an empty string it is still joined with the rest using <br/>, producing superfluous blank lines.

**Comment by lanzagar** [comment]
The empty class section was fixed, but metas were not!

**Comment by janezd** [comment]
Done. I made a separate commit, so you can see the change  -- and also so that I can revert it if that's not what we'd like. Class is treated separately (not removed even if used for axes), while metas and attributes are removed when used. 

When you confirm it's OK, I'll squash the commits.

**Comment by lanzagar** [comment]
Looks good! Merging after squash...

**Comment by janezd** [comment]
Squashed.

**Pull request merged**